lib/Runner.js

```diff
@@ -63,40 +65,38 @@ class Runner {
   }
 
   static _parseCycleResult(cycleResult) {
-    const meaningfullResults = cycleResult.slice(1); // first result is usually influenced by cold start
+    const percentiles = PERCENTILES.reduce((memo, i) => {
+      return Object.assign(memo, { [i]: percentile(i, cycleResult) });
+    }, {});
 
     return {
-      min: Math.min(...meaningfullResults),
-      max: Math.max(...meaningfullResults),
-      avg: Runner._getAverage(meaningfullResults)
+      min: Math.min(...cycleResult),
+      max: Math.max(...cycleResult),
+      avg: Runner._getAverage(cycleResult),
+      percentiles
     };
   }
 
-  _cycle(count, acc) {
-    if (count <= 0) {
-      return BPromise.resolve(acc);
-    }
-
-    if (!acc) {
-      acc = []; // eslint-disable-line no-param-reassign
-    }
-
-    const self = this;
-    return BPromise
-      .delay(INVOCATION_DELAY)
+  _cycle(count) {
+    return BPromise.resolve()
       .then(() => {
-        return typeof self.payload === 'function'
-          ? self.payload(acc.length + 1)
-          : self.payload;
+        // warm-up
+        return this._invokeLambda(0);
       })
-      .then(jsonPayload => {
-        return self._invokeLambda(jsonPayload)
-          .then(invocationDuration => {
-            acc.push(invocationDuration);
-
-            console.log('Invocation time: %dms', invocationDuration);
-            return self._cycle(count - 1, acc);
-          });
+      .delay(500)
```

Comment:
could make it longer, or maybe even wait longer for less MB (since it takes longer to cold start function)